addok_france/utils.py

```diff
@@ -160,10 +160,15 @@ def add(labels, label):
 def match_housenumber(helper, result):
     if not helper.check_housenumber:
         return
+    tokens = []
     for token in sorted(helper.tokens, key=lambda t: t.position):
-        if token.kind == 'housenumber' and token in result.housenumbers:
-            data = result.housenumbers[str(token)]
-            result.housenumber = data.pop('raw')
-            result.type = 'housenumber'
-            result.update(data)
-            break
+        if token.kind == 'housenumber':
```

Comment:
Not exactly. :)
First point is that we don't want to rely on position in addok core, because address format is so different from one country to another.
But then in addok-france, we may deal with it, and we sort of do: we use a "position pattern" to define if a token is a housenumber (basically a number followed by a type (rue, boulevard…)). Then we flag the token as "housenumber".
Once in "match_housenumber", we only reorder the tokens by their original positions, and we group the ones flagged as "housenumber".
This "grouping" operation is not the cleanest piece of code ever, I must admit. But given a housenumber can be "multitokens", that's the simplest I found :s